controllers/managedocs_controller.go

```diff
@@ -693,14 +694,9 @@ func (r *ManagedOCSReconciler) reconcileAlertmanagerConfigSecret() error {
 			return fmt.Errorf("DeadMan's Snitch secret does not contain a SNITCH_URL entry")
 		}
 
-		alertmanagerConfig := r.generateAlertmanagerConfig(pagerdutyServiceKey, dmsURL)
-		config, err := yaml.Marshal(alertmanagerConfig)
-		if err != nil {
-			return fmt.Errorf("Unable to encode alertmanager conifg: %v", err)
-		}
-		r.alertmanagerConfigSecret.Data = map[string][]byte{
-			"alertmanager.yaml": config,
-		}
+		desired := templates.AlertmanagerConfigTemplate.DeepCopy()
+		r.updateAlertmanagerConfig(desired, pagerdutyServiceKey, dmsURL)
```

Comment:
Why do we need to call into a separate func where the entire body of that func is 3 or 4 lines of code?
It just complicates reading this func because it makes the reader jump between functions with no real reason.

Can we just embed the needed changes to the desired state here?